bug: AnnotationWorker が find_image_ids_by_phashes を重複呼び出し (N+1 防止違反)

## 概要

`AnnotationWorker` 内で `find_image_ids_by_phashes` が 2 箇所で呼ばれており、N+1 クエリ防止ポリシーに違反している。テストは単一呼び出しを期待している。

## 失敗テスト

- `tests/integration/gui/workers/test_annotation_worker.py::test_annotation_worker_no_duplicate_phash_query`

## エラー内容

\`\`\`
AssertionError: find_image_ids_by_phashes was called 2 times, expected 1
\`\`\`

## 根本原因

`src/lorairo/gui/workers/annotation_worker.py:247, 295` の 2 箇所で同じクエリを実行している。

\`\`\`python
# 247行
image_ids = self.db_manager.repository.find_image_ids_by_phashes(set(merged_results.keys()))
# ...処理...
# 295行
image_ids = self.db_manager.repository.find_image_ids_by_phashes(set(merged_results.keys()))  # 重複
\`\`\`

## 修正方針

- 最初の呼び出し結果を変数に保持し、2 回目の呼び出しを削除する
- 必要に応じて処理順序を見直し、1 回のクエリで完結するように構造を整える

\`\`\`python
image_ids = self.db_manager.repository.find_image_ids_by_phashes(set(merged_results.keys()))
# 以降は image_ids を再利用
\`\`\`

## 関連ファイル

- `src/lorairo/gui/workers/annotation_worker.py:247, 295`
- `tests/integration/gui/workers/test_annotation_worker.py`

## 参考

- `.claude/rules/logging.md` - N+1 防止の一般方針
- 過去 Issue #108 (annotation worker レイヤー修正)

--- a/src/lorairo/gui/workers/annotation_worker.py
+++ b/src/lorairo/gui/workers/annotation_worker.py
@@ -240,14 +240,12 @@ class AnnotationWorker(LoRAIroWorkerBase["AnnotationExecutionResult"]):
             )
             self._check_cancellation()
 
-            db_save_success, db_save_skip, image_summaries = self._save_results_to_database(merged_results)
-
-            # モデル統計とpHash→ファイル名マッピングを構築
-            model_statistics = self._build_model_statistics(merged_results)
-            phash_to_image_id = self.db_manager.repository.find_image_ids_by_phashes(
-                set(merged_results.keys())
+            db_save_success, db_save_skip, image_summaries, phash_to_filename = (
+                self._save_results_to_database(merged_results)
             )
-            phash_to_filename = self._build_phash_to_filename_map(phash_to_image_id)
+
+            # モデル統計を構築
+            model_statistics = self._build_model_statistics(merged_results)
 
             self._report_progress(
                 100,
@@ -277,14 +275,14 @@ class AnnotationWorker(LoRAIroWorkerBase["AnnotationExecutionResult"]):
 
     def _save_results_to_database(
         self, results: PHashAnnotationResults
-    ) -> tuple[int, int, list[ImageResultSummary]]:
+    ) -> tuple[int, int, list[ImageResultSummary], dict[str, str]]:
         """アノテーション結果をDBに保存
 
         Args:
             results: PHashAnnotationResults (phash → model_name → UnifiedResult)
 
         Returns:
-            (DB保存成功件数, スキップ件数, 画像ごとの結果概要リスト) のタプル。
+            (DB保存成功件数, スキップ件数, 画像ごとの結果概要リスト, phash→ファイル名マップ) のタプル。
 
         Note:
             ライブラリが返したpHashをfind_image_ids_by_phashesで一括DB照会。
@@ -294,7 +292,7 @@ class AnnotationWorker(LoRAIroWorkerBase["AnnotationExecutionResult"]):
         # 事前一括取得: pHash → image_id（N+1回避）
         phash_to_image_id = self.db_manager.repository.find_image_ids_by_phashes(set(results.keys()))
 
-        # pHash → ファイル名マッピング構築（結果概要用）
+        # pHash → ファイル名マッピング構築（結果概要用・呼び出し元で再利用）
         phash_to_filename = self._build_phash_to_filename_map(phash_to_image_id)
 
         # 事前一括取得: モデル名・タグ文字列を収集
@@ -346,7 +344,7 @@ class AnnotationWorker(LoRAIroWorkerBase["AnnotationExecutionResult"]):
                 logger.error(f"保存失敗 phash={phash[:8]}...: {e}", exc_info=True)
 
         logger.info(f"DB保存完了: {success_count}/{len(results)}件成功")
-        return success_count, skip_count, image_summaries
+        return success_count, skip_count, image_summaries, phash_to_filename
 
     def _build_phash_to_filename_map(self, phash_to_image_id: dict[str, int]) -> dict[str, str]:
         """pHashからファイル名へのマッピングを構築する。

